### DISCUSSION TITLE
Challenge 1: NFT transaction reverted

### DISCUSSION THREAD

**Discussion by ericramirez** [Cyfrin/foundry-full-course-cu#2789] [Q&A] [answered]
I'm getting the following error message after click on the Write in the solveChallenge method:

`
Transaction has been reverted by the EVM: { "blockHash": "0xd426af16e0abcc01765f3588c0e9e12e597c9cd37390384990c1a4ad3949ec90", "blockNumber": 6985678, "contractAddress": null, "cumulativeGasUsed": 1149737, "effectiveGasPrice": 3581117777, "from": "0x1b3a608334d6b6c0d20497fcba2c8b4a89b6171e", "gasUsed": 32935, "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "status": false, "to": "0x25056312685339b49e1d1c5a0b72ff9eff13af77", "transactionHash": "0x379e73c6e40d8c81e94b245b7d6f030f9c8ae78265c8f828c0f7c01e2dca734b", "transactionIndex": 10, "type": "0x2", "events": {} }
`

Here's the transaction hash: https://sepolia.etherscan.io/tx/0x83e74e0dde933e197c416279eb07ef4eec439a903e556018041dc790bc8b19bc

**Comment by EngrPips** [comment]
Hello @ericramirez, Please send the link to the challenge contract you are trying to solve.

**Reply by ericramirez** [reply]
This is it: https://sepolia.etherscan.io/address/0x25056312685339b49e1d1C5a0b72Ff9eff13AF77

**Reply by EngrPips** [reply]
You have to call the `solveChallenge` method with your Twitter handle. Did you do that?

**Reply by ericramirez** [reply]
Of course!

I did it a while ago, but I can't find the NFT.

**Reply by EngrPips** [reply]
Why did you call the `transferOwnership` function on the contract?

<img width="1438" alt="Screenshot 2024-11-01 at 15 11 59" src="https://github.com/user-attachments/assets/3c5f313f-f3c7-4dad-9b3a-97caf5ce693c">

That is what this transaction hash https://sepolia.etherscan.io/tx/0x83e74e0dde933e197c416279eb07ef4eec439a903e556018041dc790bc8b19bc that you shared shows.

**Reply by ericramirez** [reply]
I'm sorry, I copied the wrong one: https://sepolia.etherscan.io/tx/0x379e73c6e40d8c81e94b245b7d6f030f9c8ae78265c8f828c0f7c01e2dca734b

**Reply by EngrPips** [reply]
> I'm sorry, I copied the wrong one: https://sepolia.etherscan.io/tx/0x379e73c6e40d8c81e94b245b7d6f030f9c8ae78265c8f828c0f7c01e2dca734b

This transaction failed probably because you don't have enough ether to cover the gas fee. Please ensure you have sufficient ether for the gas fee, then retry.

**Reply by ericramirez** [reply]
I have 0.3728 SepoliaETH. Isn't it more than enough?

**Reply by EngrPips** [reply]
Well I am not sure, Maybe retry and see what it cost to solve the challenge.

**Reply by ericramirez** [reply]
![image](https://github.com/user-attachments/assets/379817bd-954c-4718-933d-bde6754bd325)

**Reply by EngrPips** [reply]
> ![image](https://private-user-images.githubusercontent.com/3688694/382438113-379817bd-954c-4718-933d-bde6754bd325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NjQ2MDIsIm5iZiI6MTczMDU2NDMwMiwicGF0aCI6Ii8zNjg4Njk0LzM4MjQzODExMy0zNzk4MTdiZC05NTRjLTQ3MTgtOTMzZC1iZGU2NzU0YmQzMjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDJUMTYxODIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWIyZWY1NDI3MGNkNmE1YjRhNmE2NDNkNzNiZTQxMmVlODA5ODY5NTBkOWQ4YTc4ZjNmYmFiMDMyNjY4ODgzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.XnN33wUf-Sbfgkb-_a4tJnCHb4qR3ZCZ_GwzGrFQh6I)

This seems like you are not calling the `solveChallenge` function correctly.

**Reply by ericramirez** [reply]
Why? How can I call it the correctly?

**Reply by EngrPips** [reply]
How many times have attempted this challenge? I am sure it is more than once.

**Reply by ericramirez** [reply]
Multiple times and I did mint the NFT a couple of months ago.

**Reply by EngrPips** [reply, marked as answer]
If you successfully solve the challenge once with an address you can't use that same address to redo the challenge it will always fail.

**Reply by ericramirez** [reply]
Understood, thanks

**Reply by EngrPips** [reply]
You are welcome.
